src/mpi/coll/transports/gentran/gentran_utils.c

```diff
@@ -275,17 +275,17 @@ int MPII_Genutil_progress_hook(int *made_progress)
     in_genutil_progress = 1;
 
     if (made_progress)
-        *made_progress = FALSE;
+        *made_progress = false;
```

Comment:
is the `made_progress` argument really optional? it would be nice to remove these branches and extra `progress` variable below if we can.

Reply:
The progress hook function is registered and maintained by src/util/mpir_progress_hook.c. By glancing through the code, it seems that you are right that made_progress argument is never a null pointer. 

However, the original Akhil's Gentran code was copied from MPIR_sched (which does the same checking): 

```
int MPIDU_Sched_progress_state(struct MPIDU_Sched_state *state, int *made_progress)
{
    int mpi_errno = MPI_SUCCESS;
    size_t i;
    struct MPIDU_Sched *s;
    struct MPIDU_Sched *tmp;
    if (made_progress)
        *made_progress = FALSE;

```

If you can confirm made_progress is always non-NULL, I will add a commit to remove that branch.